app/models/reservation_validations.rb

```diff
@@ -45,6 +45,23 @@ def available
       "full time period requested.\n")
   end
 
+  def check_consecutive
+    return unless reserved? || requested?
+    return unless equipment_model
+    return unless equipment_model.max_checkout_length
+
+    # get all reservations of same user with same equipment model
+    # that are consecutive to this reservation
+    res = Reservation.for_reserver(reserver).for_eq_model(equipment_model)
+          .consecutive_with(self)
+
+    res.each do |r|
+      next unless duration + r.duration > equipment_model.max_checkout_length
```

Comment:
Hmmm, we should probably turn this into an in-person discussion on Monday, but at first glance it seems like your first thought is beyond scope - if someone wants to miss a day here and there (meaning that the equipment spends at least one night outside of their possession), then we don't need to worry about it.

In terms of a recursion mess, I feel like approaching this with a combination of some logic based on start/due dates and status coupled with sorting the reservations by start date should avoid any major issues. There's probably a really elegant way to figure this out but I can't think of it off the top of my head.

Finally, to your last point, we basically already do this and I'm not sure that it's a good idea to do it automatically. If someone wants to reserve equipment for longer than normally allowed they can already issue a request since that validation is a cart validation, not a "hard" ActiveRecord one. That said, we might run into trouble in a case where they're trying to create an overlapping reservation that respects the length limits - in that case we'd fail the hard validation but wouldn't be given the option of issuing a request. Based on that fact, why don't we move this into the cart validations instead? That way we're still protected during creation but allow for requests to be issued. Thoughts?